engine/src/main/java/org/terasology/rendering/dag/StateChange.java

```diff
@@ -16,7 +16,14 @@
 package org.terasology.rendering.dag;
 
 /**
- * TODO: Add javadocs
+ * A StateChange denotes a change in the OpenGL state, that is required by a particular Node.
```

Comment:
I'd rewrite:

`A StateChange instance encapsulate a change in the OpenGL state that is required by one or more consecutive nodes.`